docs/source/tutorials/using_text.rst

```diff
@@ -0,0 +1,197 @@
+
+* The keys contain words or character which should be coloured seperately
+  and the values should be the color from :class:`~.Color`.
+
+For example,
+
+.. manim:: Textt2cExample
+    :save_last_frame
+
+    class Textt2cExample(Scene):
+        def construct(self):
+            t2cindices = Text('Hello', t2c={'[1:-1]': BLUE}).move_to(LEFT)
+            t2cwords = Text('World',t2c={'rl':RED}).next_to(t2cindices, RIGHT)
+            self.add(t2cindices)
+
+If you want avoid problems when colours(due to ligatures), consider using
+:class:`MarkupText`.
+
+
+Using Gradients
+---------------
+
+You can use Gradient using :attr:`~.Text.gradient`. The value must
+be a Iterable of any Length.
+
+For example,
+
+.. manim:: GradientExample
+    :save_last_frame:
+
+    class GradientExample(Scene):
+        def construct(self):
+            t = Text("Hello", gradient=(RED, BLUE, GREEN)).scale(2)
+            self.add(t)
+
+You can also use :attr:`~.Text.t2g` for using gradients with specific 
+characters of the Text. It has a very similar syntax like 
+:ref:`Using Colors`.
+
+For example,
+
+.. manim:: t2gExample
+    :save_last_frame
+
+    class t2gExample(Scene):
+        def construct(self):
+            t2gindices = Text(
+                'Hello',
+                t2g={
+                    '[1:-1]': (RED,GREEN),
+                },
+            ).move_to(LEFT)
+            t2gwords = Text(
+                'World',
+                t2g={
+                    'World':(RED,BLUE),
+                },
+            ).next_to(t2gindices, RIGHT)
+            self.add(t2gindices, t2cwords)
+
+
+.. _disable-ligatures:
+
+Disabling Ligatures
+-------------------
+
+By disabling ligatures you would get a 1-1 mapping between characters and
+submobjects. This would fix coloring issue's. 
+
+
+.. warning::
+
+    Be aware that using this method with a text which heavily needs
+    ligatures may not work as expected. For example, when disabling
+    ligatures with Arabic text the output doesn't looks as expected.
+
+You can disable ligatures by passing ``disable_ligatures`` parameter to 
+:class:`Text`. For example,
+
+.. manim:: DisableLigature
+
+    class DisableLigature(Scene):
+        def construct(self):
+            li = Text("fl ligature").scale(2)
+            nli = Text("fl ligature", disable_ligatures=True).scale(2)
+            self.add(Group(li, nli).arrange(DOWN, buff=.8))
+
+.. _iterating-text:
+
+Iterating :class:`~.Text`
+-------------------------
+
+Text objects behave like a VGroup-like iterable of all characters in the given
+text. In particular, slicing is possible.
+
+For example, you can set each letter to different color by iterating it.
+
+.. manim:: IterateColor
+
+    class IterateColor(Scene):
+        def construct(self):
+            text = Text("Colors").scale(2)
+            for letter in text:
+                letter.set_color(random_bright_color())
+                self.play(Write(letter))
```

Comment:
```suggestion
            self.add(text)
```
This animation is really long and can be archived far easier by using `Write(text, lag_ratio=1)` using a long runtime.
Also I don't think it needs to be animated.

Reply:
Done https://github.com/ManimCommunity/manim/pull/1510/commits/2b2642ce81e79fa60b50144350eafcf883b35bec